src/Compilers/VisualBasic/Portable/Emit/SynthesizedPrivateImplementationDetailsSharedConstructor.vb

```diff
@@ -0,0 +1,112 @@
+
+Imports System.Collections.Generic
+Imports System.Collections.Immutable
+Imports System.Diagnostics
+Imports System.Linq
+Imports Microsoft.CodeAnalysis.CodeGen
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
+    Friend NotInheritable Class SynthesizedPrivateImplementationDetailsSharedConstructor
+        Inherits SynthesizedGlobalMethodBase
+
+        Private ReadOnly _containingModule As SourceModuleSymbol
+        Private ReadOnly _privateImplementationType As PrivateImplementationDetails
+        Private ReadOnly _voidType As TypeSymbol
+
+        Friend Sub New(
+            containingModule As SourceModuleSymbol,
+            privateImplementationType As PrivateImplementationDetails,
+            voidType As NamedTypeSymbol
+        )
+            MyBase.New(containingModule, WellKnownMemberNames.StaticConstructorName, privateImplementationType)
+
+            _containingModule = containingModule
+            _privateImplementationType = privateImplementationType
+            _voidType = voidType
+        End Sub
+
+        Public Overrides ReadOnly Property IsSub As Boolean
+            Get
+                Return True
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property ReturnsByRef As Boolean
+            Get
+                Return False
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property ReturnType As TypeSymbol
+            Get
+                Return _voidType
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property HasSpecialName As Boolean
+            Get
+                Return True
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property MethodKind As MethodKind
+            Get
+                Return MethodKind.SharedConstructor
+            End Get
+        End Property
+
+        Friend Overrides Function GetBoundMethodBody(compilationState As TypeCompilationState, diagnostics As DiagnosticBag, Optional ByRef methodBodyBinder As Binder = Nothing) As BoundBlock
+            methodBodyBinder = Nothing
+
+            Dim factory As New SyntheticBoundNodeFactory(Me, Me, VisualBasicSyntaxTree.Dummy.GetRoot(), compilationState, diagnostics)
+
+            ' Initialize the payload root for for each kind of dynamic analysis instrumentation.
+            ' A payload root Is an array of arrays of per-method instrumentation payloads.
+            ' For each kind of instrumentation:
+            '
+            '     payloadRoot = New T(MaximumMethodDefIndex)() {}
+            '
+            ' where T Is the type of the payload at each instrumentation point, And MaximumMethodDefIndex Is the 
+            ' index portion of the greatest method definition token in the compilation. This guarantees that any
+            ' method can use the index portion of its own method definition token as an index into the payload array.
```

Comment:
OK. As you probably guessed, I copied the C# code and then edited it to make it VB. In the process, VB attempted to format "comments" beginning with // as code.